### PR TITLE
Custom Pie Chart Labels

### PR DESCRIPTION
**Pull request by joneschrisan**
Added working custom pie chart labels

### CODE REVIEW

File: src/models/pie.js [hunk 1]
```diff
@@ -298,16 +298,20 @@ nv.models.pie = function() {
...
-                                label = getX(d.data);
-                                break;
-                            case 'value':
-                                label = valueFormat(getY(d.data));
-                                break;
-                            case 'percent':
-                                label = valueFormat(percent);
-                                break;
+                        if(typeof labelType === 'function') {
+                            label = labelType({ 'key': getX(d.data), 'value': getY(d.data), 'percent': labelFormat(percent) });
```
Comment (liquidpele) [inline src/models/pie.js]: Just pass d to it and let the function handle everything in this case.

**Comment by liquidpele** [comment]
This looks good besides that one thing!

**Comment by joneschrisan** [comment]
But keeping the other 3 hard coded labels in there would mean backward compatibility. Otherwise everyone will have to change their labels to anonymous functions.

**Comment by joneschrisan** [comment]
I tried passing d to it but it couldn't run the other functions getX, getY, and labelFormat as they had gone out of scope.

**Comment by liquidpele** [comment]
Shouldn't labelFormat be valueFormat there?   Also you can get to it from the chart options... e.g:

    chart.labelType(function(d) {
        var key = chart.x()(d.data);
        var value = chart.y()(d.data);
        var percent = chart.valueFormat()(percent);
        return key + ':' + value + ':' + percent;
    });

**Comment by liquidpele** [comment]
Or if you want, you can pass the object  you're passing right now, but just add an attribute 'data' that is d... mostly I just want to make sure all available data is passed *somewhere* there... :)

**Comment by joneschrisan** [comment]
It should yes, I just noticed that too. I'd agree with you, just didn't know a way and needed to get it working quick. Passing d and then using that would be better, more usable.

**Comment by liquidpele** [comment]
ping.  can you put d into the object you pass with the other params, just so the raw data object is available to the function?

**Comment by liquidpele** [comment]
@joneschrisan you still around?

**Comment by robinfhu** [comment]
I'll take care of this

**Pull request merged**